libraries/app/application.cpp

```diff
@@ -997,6 +1016,7 @@ void application::set_program_options(boost::program_options::options_descriptio
    configuration_file_options.add_options()
          ("p2p-endpoint", bpo::value<string>(), "Endpoint for P2P node to listen on")
          ("p2p-max-connections", bpo::value<uint32_t>(), "Maxmimum number of incoming connections on P2P endpoint")
+         ("p2p-parameters", bpo::value<uint32_t>(), "P2P network parameters")
```

Comment:
This should be set to some default value with the options that currently exist.

The type does also not match the parsing logic in `resest_p2p_node`.

I would make a struct for the parameters with the default values set to the defaults for the config.
For the default value, construct the default initialized struct, and convert to string.